### ISSUE TITLE
plans.js get stripe plans from config array

### ISSUE THREAD

**Issue by atherdon** [ChickenKyiv/loopback-tutor-intern-8#64]
from https://github.com/ChickenKyiv/loopback-tutor-intern-8/blob/frontend/src/utils/stripe-plans.js
to https://github.com/ChickenKyiv/loopback-tutor-intern-8/blob/frontend/src/pages/subscription/Plans/Plans.js#L8

**Comment by chauhannishith** [comment]
You have to setup plans manually in stripe after which they give you the id and then you have to pass that id to the stripe component so all other data is with stripe you only have to add one more else-if for different plans
so i don't think it is possible dynamically atleast in frontend part

**Comment by atherdon** [comment]
yep, i know it, i done it before, year ago. this is why i wanted to have
one place, where plans will be stored. maybe you got me wrong. you can move
your ids, with pricing into config file. just don't want to have passing
plans data into constuctor. it's not very clear


On 31-May-2018 at 05:12 AM, chauhannishith wrote:


You have to setup plans manually in stripe after which they give you the id
and then you have to pass that id to the stripe component so all other data
is with stripe you only have to add one more else-if for different plans
so i don't think it is possible dynamically atleast in frontend part

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-393405677>,
or mute the thread

**Comment by chauhannishith** [comment]
Will try to find a solution which solves this problem, something like passing an array

**Comment by atherdon** [comment]
you son't need to find a new, custom way. in my prev repo - i think i
shared it with you - i made such sysmte. we just need to move it genlty to
react. was you able to review that code?


On 31-May-2018 at 11:51 AM, chauhannishith wrote:


Will try to find a solution which solves this problem, something like
passing an array

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-393505646>,
or mute the thread

**Comment by chauhannishith** [comment]
I thought that you had created that file to generate plans dynamically as the content was like that

**Comment by atherdon** [comment]
maybe, in future. but it'll require calls to stripe to get actual plans
data.


On 31-May-2018 at 12:15 PM, chauhannishith wrote:


I thought that you had created that file to generate plans dynamically as
the content was like that

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-393510958>,
or mute the thread

**Comment by atherdon** [comment]
check this project, not sure if i share it with you before: https://github.com/ChickenKyiv/stripe-recurring-membership

maybe it'll help you

**Comment by atherdon** [comment]
yep, different, but working version can give you hints...


On 01-Jun-2018 at 11:23 AM, chauhannishith wrote:


Yes i did see this but as you said that you wanted in react so that was a
different thing altogether and that's why i did it in the way that i found

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-393851983>,
or mute the thread

**Comment by atherdon** [comment]
Hi Nishith, please postpone work, related to stripe configuration for a few days, i think i find some crazy idea, that will help us! or maybe i just need to go sleep :)

**Comment by atherdon** [comment]
i have a question. for your stripe pricing plans, did you create/activate stripe account?
it's a bit confusing for me. one year ago, when i was working on it = it has a different interface/logic...

It looks like they add a lot of new features, but previous working stuff - disabled

**Comment by chauhannishith** [comment]
No, I haven't activated my account as business, I think you need to get the tax id and business related information for that

**Comment by chauhannishith** [comment]
> It looks like they add a lot of new features, but previous working stuff - disabled

yes they have changed up a lot, when i first tried my hand at it it required me to create a for and send the token myself with the data but now it is handling all that itself

**Comment by atherdon** [comment]
strange... i understand that they are growing and making all is better.
but before it was simple for developers at first point. so code written
1year ago with working checkout is not working anymore. i start to think
about not including stripe...


On 02-Jun-2018 at 05:11 AM, chauhannishith wrote:


It looks like they add a lot of new features, but previous working stuff -
disabled

yes they have changed up a lot, when i first tried my hand at it it
required me to create a for and send the token myself with the data but now
it is handling all that itself

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-394059396>,
or mute the thread

**Comment by atherdon** [comment]
will reply when wake up.


On 02-Jun-2018 at 05:11 AM, chauhannishith wrote:


It looks like they add a lot of new features, but previous working stuff -
disabled

yes they have changed up a lot, when i first tried my hand at it it
required me to create a for and send the token myself with the data but now
it is handling all that itself

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/loopback-tutor-intern-8/issues/64#issuecomment-394059396>,
or mute the thread
